packages/web3-eth-admin/types/index.d.ts

```diff
@@ -0,0 +1,91 @@
+/*
+    This file is part of web3.js.
+    web3.js is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    web3.js is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+    You should have received a copy of the GNU Lesser General Public License
+    along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @file index.d.ts
+ * @author Prince Sinha <sinhaprince013@gmail.com>
+ * @date 2019
+ */
+
+import {Accounts} from 'web3-eth-accounts';
+import {provider} from 'web3-providers';
+import {AbstractWeb3Module, Web3ModuleOptions, NodeInfo} from 'web3-core';
+import * as net from 'net';
+
+export class Admin extends AbstractWeb3Module {
+    constructor(provider: provider, net?: net.Socket|null, options?: Web3ModuleOptions, accounts?: Accounts|null);
+
+    addPeer(
+        url: string,
+        callback?: (error: Error, result: boolean) => void
+    ): Promise<boolean>;
+
+    getDataDirectory(
+        callback?: (error: Error, result: string) => void
+    ): Promise<string>;
+
+    getNodeInfo(
+        callback?: (error: Error, result: NodeInfo) => void
+    ): Promise<NodeInfo>;
+
+    getPeers(
+        callback?: (error: Error, result: any[]) => void
+    ): Promise<any[]>;
+
+    setSolc(
+        path: string,
+        callback?: (error: Error, result: string) => void
+    ): Promise<string>;
+
+    startRPC(callback?: (error: Error, result: boolean) => void): Promise<boolean>
```

Comment:
These types don't go well due to 4 optional parameters which makes it impossible to make this nice typing's. If you repeat stuff like 

```
startRPC(host: string, callback?: (error: Error, result: boolean) => void): Promise<boolean>;
startRPC(cors: string, callback?: (error: Error, result: boolean) => void): Promise<boolean>;
```

in TS it wont know which type to resolve to if you await them as both accept a string. 

Sorry to be a pain please revert back to what you had before as this wont work at all. I think 1 type with 4 optional parameters is the only way here unless you broke the methods down which i thinks overkill. You can also revert my tslint change and remove the `"unified-signatures": false` line.

Reply:
okay so, I will revert it as it was before @joshstevens19

Reply:
But If I will revert it to as it was before then, I will not able to test all the outcomes @joshstevens19.

```
    startRPC(
        host?: string,
        port?: number,
        cors?: string,
        apis?: string,
        callback?: (error: Error, result: boolean) => void
    ): Promise<boolean>;
```
consider this test:

```
// $ExpectType Promise<boolean>
admin.startRPC(8545);
```
We will not able to test this.

Reply:
yes hence why i said put `null` in the parameters you need to ignore. They are all optional parameters so we need to cover each case really i.e.

```
// $ExpectType Promise<boolean>
admin.startRPC(null, 8546);
```

Reply:
Argument of type 'null' is not assignable to parameter of type 'string | undefined'.ts(2345).

Reply:
For that , I need to add "strictNullChecks": false.